### PR TITLE
MP/ Add business and address gql interfaces

### PR DESCRIPTION
**Pull request by MPucheta**

### CODE REVIEW

File: src/controllers/business/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      CUIT: Number(CUIT),
+      AFIPCondition,
+      physicalAddressId: address.id
+    }, { transaction })
+
+    const business = await models.business.create({
+      fantasyName,
+      physicalAddressId: address.id
+    }, { transaction })
+
```
Comment (MPucheta) [inline src/controllers/business/index.js]: I though about putting the physicalAddress to be allowed Null when creating the instances, and then set them up as `setPhysicalAddress` as I did with the legalInformation, but it didn't seem too beneficial, what do you think?

Comment (nahudalla) [reply src/controllers/business/index.js]: You can create everything in a single operation: https://sequelize.org/master/manual/creating-with-associations.html

File: src/controllers/business/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    await business.setLegalInformation(legalInformation, { transaction })
+
+    return business
+  })
+}
+
+export const getBusinessById = (obj, { id }, { models }) => models.business.findByPk(id)
```
Comment (nahudalla) [inline src/controllers/business/index.js]: You're returning `null` if the given `id` doesn't exist in the database, but in GraphQL you defined the query to be not `null`.

You should either:
- throw an error in this resolver if a business was not found with the given `id`, or
- change the schema to be able to return `null`.

I prefer the first option, throwing an error that contains an error code (to be able to distinguish it in the front-end) and a descriptive message.

**Review by nahudalla — CHANGES_REQUESTED** [review]

File: src/schema/types/business.graphql [hunk 3]
```diff
@@ -0,0 +1,31 @@
+type Business {
+  id: ID!
+  fantasyName: String!
+  realName: String!
+  vertical: String!
+  CUIT: String!
+  AFIPCondition: AFIPConditionEnum!
+  address: Address!
+}
```
Comment (roke77) [inline src/schema/types/business.graphql]: I agree with this implementation, but just for being curious, what do you guys think about extracting the Legal Information into its own type? Wouldn't this simplify the Business controller resolvers?

```suggestion
type LegalInformation {
  fantasyName: String!
  realName: String!
  vertical: String!
  CUIT: String!
  AFIPCondition: AFIPConditionEnum!
}

type Business {
  id: ID!
  legalInformation: LegalInformation!
  address: Address!
 }
```

Comment (MPucheta) [reply src/schema/types/business.graphql]: I think it shows the underlying structure too much, it makes the implementation easier though.

Comment (nahudalla) [reply src/schema/types/business.graphql]: Yeah, I agree with Mauro. If business rules dictate that a business can have only one legal information, then I think the solution implemented by Mauro is better for usage by a client.

**Review by roke77 — APPROVED** [review]

**Review by nahudalla — APPROVED** [review]

**Pull request merged**